src/CancelRequestDialog.test.js

```diff
@@ -0,0 +1,113 @@
+import {
+  fireEvent,
+  render,
+  screen,
+} from '@testing-library/react';
+
+import '../test/jest/__mock__';
+
+import CancelRequestDialog from './CancelRequestDialog';
+
+const labelIds = {
+  modalLabel: 'ui-requests.cancel.modalLabel',
+  modalText: 'ui-requests.cancel.requestWillBeCancelled',
+  cancelRequestDialogCancel: 'stripes-core.button.confirm',
+  cancelRequestDialogClose: 'stripes-core.button.back',
+};
+const testIds = {
+  cancelRequestDialog: 'cancelRequestDialog',
+  cancelRequestDialogCancel: 'cancelRequestDialogCancel',
+  cancelRequestDialogClose: 'cancelRequestDialogClose',
+};
+
+describe('CancelRequestDialog', () => {
+  const onCancelRequest = jest.fn();
+  const onClose = jest.fn();
+  const cancellationReasons = {
+    records: [{
+      name: 'cancellation reasons',
+      requiresAdditionalInformation: false,
+    }, {
+      name: 'other cancellation reasons',
+      requiresAdditionalInformation: true,
```

Comment:
Could you add tests to assert if additional information placeholder is displayed based on "cancellationReasons"?